packages/bun-usockets/src/crypto/openssl.c

```diff
@@ -457,7 +460,8 @@ struct us_internal_ssl_socket_t *ssl_on_data(struct us_internal_ssl_socket_t *s,
     // we need to check if we received a shutdown here
     if (SSL_get_shutdown(s->ssl) & SSL_RECEIVED_SHUTDOWN) {
       s->received_ssl_shutdown = 1;
-      // we will only close after we handle the data and errors
+      us_internal_ssl_socket_close(s, 0, NULL);
```

Comment:
wouldn't this cause data to be lost?

Reply:
yeah we need to drain the buffer first, just fixed it rn